### PR TITLE
Playback priority

### PR DESCRIPTION
**Pull request by codealchemist**
Added highest playback priority feature; added highest playback priority checkbox on preferences.
Pauses all active torrents when playback starts, giving all bandwidth to the streaming torrent.
Restores paused torrent when playback stops.
This is the _easy way_ to solve #362.

### CODE REVIEW

**Comment by noamokman** [comment]
I think we need to resume downloads when the torrent you are watching finished downloading.
It looks like it only resumes after you close the player, am I right?

**Comment by codealchemist** [comment]
Good catch @noamokman!
Yeah, on this PR the paused downloads resume only after the player stops.
It will be better if it also resumes when the streaming torrent finishes downloading.
I'll look into it, thanks!

**Comment by codealchemist** [comment]
@noamokman, now it resumes on `wt-done`, which seems to be working as expected.

**Comment by noamokman** [comment]
Awesome! Great work :smile:

**Comment by codealchemist** [comment]
Thanks :)

**Comment by codealchemist** [comment]
@feross if this is still needed I can spend some time fixing conflicts.
Thx.

**Comment by toinouu** [comment]
Exactly what I want 
+1

**Comment by codealchemist** [comment]
Thanks @toinouu!
Still waiting for team approval before fixing conflicts and merge.
:)

**Comment by dcposch** [comment]
@codealchemist Sorry we missed this. Reviewing today

**Comment by codealchemist** [comment]
No worries @dcposch :)
Thanks for reviewing it!

File: src/renderer/controllers/torrent-list-controller.js [hunk 1]
```diff
@@ -103,18 +103,73 @@ module.exports = class TorrentListController {
     }
   }
 
+  pauseAll ({filter, excluded}) {
```
Comment (dcposch) [inline src/renderer/controllers/torrent-list-controller.js]: What about making this
```
prioritizeTorrent (infohash) {
```

...the filter arg is always just active torrents. Don't know if we need the ability to pause torrents by arbitrary JSON filter. The excluded arg is always an array containing a single infohash: the torrent we're playing.

**Comment by codealchemist** [comment]
Thanks a lot for your review @dcposch!
I'll work on it.
Cheers!

**Comment by codealchemist** [comment]
Tests passing.
Manual test looks good.

**Comment by feross** [comment]
🎉 🎉 🎉 🎉

**Pull request merged**